### PR TITLE
Add nearest NOAA point column to sites and fill script

### PR DESCRIPTION
**Pull request by echaidemenos**
This PR resolves https://github.com/aqualinkorg/aqualink-app/issues/784

- adds column `nearest_noaa_location` of type `geometry` to `sites` table
- adds script `yarn fill-noaa-nearest-point` to fill the above column at existing sites

Nearest available NOAA point is calculated, using netcdf files from https://coralreefwatch.noaa.gov/, which contain world wide data for single dates. Then running a breadth first search on the world 2x2 grid we calculate the nearest point.

This PR is a second attempt of it's [original](https://github.com/aqualinkorg/aqualink-app/pull/788), since we had issues deploying to cloud. It would be nice though if, in the future, we could automate the process instead of running the script manually.

### CODE REVIEW

**Review by ericboucher — APPROVED** [review]

File: packages/api/scripts/noaa-availability.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  Logger.error(
+    'NetCDF is not installed. Please install NetCDF before continuing',
+  );
+  process.exit();
+}
+
+function getFileUrl() {
+  const date = new Date();
+
+  // No data are available for last 2 days
```
[the remark this answers is not in the document]
Comment (ericboucher) [reply packages/api/scripts/noaa-availability.ts:30]: Agreed. Maybe rephrase this to `// NOAA data is usually available with a 2-day lag.`

File: packages/api/src/utils/sofar-availability.ts [hunk 2]
```diff
@@ -18,5 +19,40 @@ export function getSofarNearestAvailablePoint(point: Coord): [number, number] {
...
+        // first again
+        [
+          ((180 + longitude - 0.25) % 360) - 180,
+          ((90 + latitude + 0.25) % 180) - 90,
+        ],
+      ],
+    ],
+  };
+
+  const pointCoordinates = (point as Point)?.coordinates || point;
```
[the remark this answers is not in the document]
Comment (echaidemenos) [reply packages/api/src/utils/sofar-availability.ts:53]: Nice observation. The test where passing `[number, number]` instead of `Cord`, which by definition can be `Point`

File: packages/api/scripts/noaa-availability.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+  return [
+    Number((result[0] * 0.05 - 180).toFixed(3)),
+    Number((result[1] * 0.05 - 90).toFixed(3)),
+  ];
+}
+
+async function run() {
+  const { s: sites } = argv;
+  const siteIds = sites && sites.map((site) => parseInt(`${site}`, 10));
```
[the remark this answers is not in the document]
Comment (echaidemenos) [reply packages/api/scripts/noaa-availability.ts:134]: By the default behavior of typeorm `find` if some id does not exists, it will return nothing, so this will be just ignored. If the id is invalid (ex: '13f', which is not a number) there will be an error

File: packages/api/scripts/noaa-availability.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ericboucher) [reply packages/api/scripts/noaa-availability.ts]: It's days not years in this case

[the remark this answers is not in the document]
Comment (ericboucher) [reply packages/api/scripts/noaa-availability.ts]: But since the availability zone should not change, or at least not often at all @echaidemenos you could also simply set a date that works and set that as a constant for now.

**Comment by ericboucher** [comment]
@echaidemenos it looks good overall, just a few potential changes:
- let's optimize the script to go a bit faster in future runs. For example, you could filter to only process entities that do not have a nooaAvailabilityPoint. With an "--all" flag in case we want to rerun for all sites
- let's fix a date for the availability flag, it's not worth pulling a different file every time and I don't think the zone will change anytime soon.
